amf-validation/shared/src/main/scala/amf/validation/internal/shacl/custom/CustomShaclValidator.scala

```diff
@@ -21,9 +21,12 @@ import scala.concurrent.{ExecutionContext, Future}
 
 object CustomShaclValidator {
 
-  type PropertyInfo = (Annotations, Field)
-  // When no property info is provided violation is thrown in domain element level
-  type CustomShaclFunction  = (AmfObject, Option[PropertyInfo] => Unit) => Unit
+  case class ValidationInfo(field: Field, message: Option[String] = None)
```

Comment:
Wouldn't remove the use of annotations

Reply:
Weren't being used: no `CustomShaclFunction` was passing them, and the `CustomShaclValidator` was not using them for anything. I agree they are useful, but that requires using them.

Using them requires:
- changing all custom shacl functions to pass element annotations
- changing the validator logic to use annotations for useful information (location, lexical, ...)
- this requires another refactor and will possibly will change many validation reports

We can create a story for this enhancement, but until then I'm removing obscure and unused code.